lib/Models/WebMapServiceCatalogItem.js

```diff
@@ -409,8 +409,24 @@ WebMapServiceCatalogItem.prototype.updateFromCapabilities = function(capabilitie
     }
 
     var legendUri, legendMimeType;
-    // There can be multiple styles - just get the first if so.
+    // If style is defined in parameters, use that, but only if a style with that name can be found.
```

Comment:
We don't verify that a style exists before passing it to `GetMap`, so I'm not sure why we would here.  IMO `parameters` should be passed through to the server without any judgement or interpretation, just do what the user tells us to do.  I think we should pass all of the `parameters` to `GetLegendGraphic` too, not just `style`.